pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -170,7 +170,8 @@ def model_to_device(self) -> None:
         self.model = self.wrapped_model.to(self.device)
 
     def barrier(self, name: Optional[str] = None) -> None:
-        rendezvous(name)
+        if tpu_distributed:
+            rendezvous(name)
```

Comment:
so that's the equivalent of `torch.distributed.is_avalable() and torch.distributed.is_initialized()` correct?

Reply:
It's checking the world size.

```python
    if _TPU_AVAILABLE:
        return xm.xrt_world_size() > 1
    return False
```

Reply:
should we move these out of the training type plugins and into a general utilities? we can check if TPU or if torch distributed and handle accordingly

Reply:
I like the idea of having it in Training Type plugins.